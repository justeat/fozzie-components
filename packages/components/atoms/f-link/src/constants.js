export const VALID_LINK_TYPES = ['default', 'external', 'newLocation'];
export const DEFAULT_LINK_TYPE = 'default';
