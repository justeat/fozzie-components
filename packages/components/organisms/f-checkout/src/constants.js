export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHODS = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];

export const TENANT_MAP = {
    'en-GB': 'uk',
    'en-AU': 'au',
    'en-NZ': 'nz',
    'da-DK': 'dk',
    'es-ES': 'es',
    'en-IE': 'ie',
    'it-IT': 'it',
    'nb-NO': 'no'
};

export const BASKET_NOT_ORDERABLE_ERRORS = ['RESTAURANT_NOT_TAKING_ORDERS', 'SERVICE_TYPE_UNAVAILABLE', 'ITEMS_UNORDERABLE', 'ADDITIONAL_ITEMS_REQUIRED', 'MINIMUM_ORDER_VALUE_NOT_MET', 'LOCATION_UNDELIVERABLE', 'AGE_VERIFICATION_FAILED'];
export const INVALID_MODEL_STATE_ERRORS = ['FIRST_NAME_REQUIRED', 'LAST_NAME_REQUIRED', 'PHONE_NUMBER_REQUIRED', 'DATE_OF_BIRTH_REQUIRED', 'FULFILMENT_TIME_REQUIRED', 'ADDRESS_LINES_REQUIRED', 'POSTAL_CODE_REQUIRED', 'GEO_LOCATION_REQUIRED'];
export const SET_ORDER_TIME_ERROR = 'FULFILMENT_TIME_UNAVAILABLE'

export const VALIDATIONS = {
    address: 'addressValidations',
    guest: 'guestValidations'
};

export const VUEX_CHECKOUT_MODULE = 'fCheckoutModule';
export const VUEX_CHECKOUT_ANALYTICS_MODULE = 'fCheckoutAnalyticsModule';
