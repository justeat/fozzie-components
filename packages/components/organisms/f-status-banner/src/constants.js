export const BASE_HERO_URL = 'https://d30v2pzvrfyzpo.cloudfront.net/status-banner/';
