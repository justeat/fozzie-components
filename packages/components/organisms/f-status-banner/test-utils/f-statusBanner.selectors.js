export const COMPONENT = '[data-test-id="main-banner-container"]';
export const ADDRESS_BOX = '[data-test-id="address-box-input"]';
export const BUTTON = '[data-test-id="find-restaurants-button"]';
