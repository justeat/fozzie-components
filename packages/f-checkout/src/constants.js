export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHOD = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];
