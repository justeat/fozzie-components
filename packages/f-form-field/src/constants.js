export const VALID_INPUT_TYPES = ['text', 'email', 'password', 'radio', 'checkbox'];
export const CUSTOM_INPUT_TYPES = ['dropdown'];
export const DEFAULT_INPUT_TYPE = 'text';

export const VALID_LABEL_STYLES = ['default', 'inline', 'inlineNarrow'];

export const VALID_GROUP_POSITION = ['top', 'bottom', 'middle'];

export const MOBILE_WIDTH = 768;
