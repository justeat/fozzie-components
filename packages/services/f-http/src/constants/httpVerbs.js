const METHOD_GET = 'GET';
const METHOD_POST = 'POST';

export default {
    METHOD_GET,
    METHOD_POST
};
