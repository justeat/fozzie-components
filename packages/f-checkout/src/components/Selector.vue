<template>
    <form-field
        id="time-selection"
        data-test-id="form-select"
        :class="$style['o-form-select']"
        input-type="dropdown"
        :label-text="orderMethod"
        :dropdown-options="fulfilmentTimes"
        @input="selectionChanged" />
</template>

<script>
import { mapState } from 'vuex';
import FormField from '@justeat/f-form-field';
import { CHECKOUT_METHOD_DELIVERY } from '../constants';

export default {
    components: { FormField },

    computed: {
        ...mapState('checkout', [
            'fulfillment',
            'serviceType'
        ]),

        orderMethod () {
            return this.serviceType === CHECKOUT_METHOD_DELIVERY
                ? this.$t('labels.deliveryOrderMethod')
                : this.$t('labels.collectionOrderMethod');
        },

        fulfilmentTimes () {
            const times = this.fulfillment.times.map(time => time.label.text);
            return times;
        }
    },

    methods: {
        selectionChanged (e) {
            this.fulfillment.times.forEach(el => { el.selected = false; });
            const newSelected = this.fulfillment.times.find(time => time.label.text === e);
            if (newSelected) {
                newSelected.selected = true;
            }
        }
    }
};
</script>

<style lang="scss" module>
.o-form-select {
    margin-top: spacing(x2);
}
</style>
