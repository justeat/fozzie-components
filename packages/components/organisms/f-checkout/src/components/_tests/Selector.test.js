import Vuex from 'vuex';
import { VueI18n } from '@justeat/f-globalisation';
import { shallowMount, createLocalVue, mount } from '@vue/test-utils';
import { CHECKOUT_METHOD_DELIVERY, CHECKOUT_METHOD_COLLECTION } from '../../constants';
import Selector from '../Selector.vue';
import {
    fulfilmentTimes, defaultCheckoutState, i18n, createStore
} from './helpers/setup';

const localVue = createLocalVue();

localVue.use(VueI18n);
localVue.use(Vuex);

describe('Selector', () => {
    const propsData = {};

    it('should be defined', () => {
        // Arrange & Act
        const wrapper = shallowMount(Selector, {
            store: createStore(),
            i18n,
            localVue,
            propsData
        });

        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    describe('computed ::', () => {
        describe('orderMethod', () => {
            it('should show the delivery label when the `serviceType` is `delivery`', () => {
                // Arrange & Act
                const wrapper = mount(Selector, {
                    store: createStore({ ...defaultCheckoutState, serviceType: CHECKOUT_METHOD_DELIVERY }),
                    i18n,
                    localVue,
                    propsData
                });

                const selector = wrapper.find('[data-test-id="formfield-order-time-dropdown-select"]');

                // Assert
                expect(selector.html()).toMatchSnapshot();
            });

            it('should show the collection label when the `serviceType` is `collection`', () => {
                // Arrange & Act
                const wrapper = mount(Selector, {
                    store: createStore({ ...defaultCheckoutState, serviceType: CHECKOUT_METHOD_COLLECTION }),
                    i18n,
                    localVue,
                    propsData
                });

                const selector = wrapper.find('[data-test-id="formfield-order-time-dropdown-select"]');

                // Assert
                expect(selector.html()).toMatchSnapshot();
            });
        });

        describe('fulfilmentTimes', () => {
            it('should create an array of formatted fulfilment times', () => {
                // Arrange & Act
                const wrapper = shallowMount(Selector, {
                    store: createStore({ ...defaultCheckoutState }),
                    i18n,
                    localVue,
                    propsData
                });

                const expectedTimes = [
                    { text: 'As soon as possible', value: '2020-01-01T01:00:00.000Z' },
                    { text: 'Wednesday 01:15', value: '2020-01-01T01:15:00.000Z' }
                ];

                // Assert
                expect(wrapper.vm.fulfilmentTimes).toEqual(expectedTimes);
            });

            it('should create an array of formatted fulfilment times without `as soon as possible` if `isAsapAvailable` is `false`', () => {
                // Arrange & Act
                const wrapper = shallowMount(Selector, {
                    store: createStore({
                        ...defaultCheckoutState,
                        availableFulfilment: {
                            times: fulfilmentTimes,
                            isAsapAvailable: false
                        }
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                const expectedTimes = [
                    { text: 'Wednesday 01:00', value: '2020-01-01T01:00:00.000Z' },
                    { text: 'Wednesday 01:15', value: '2020-01-01T01:15:00.000Z' }
                ];

                // Assert
                expect(wrapper.vm.fulfilmentTimes).toEqual(expectedTimes);
            });

            it('should create an empty array when there are no times available and `isAsapAvailable` is `false`', () => {
                // Arrange & Act
                const wrapper = shallowMount(Selector, {
                    store: createStore({
                        ...defaultCheckoutState,
                        availableFulfilment: {
                            times: [],
                            isAsapAvailable: false
                        }
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                const expectedTimes = [];

                // Assert
                expect(wrapper.vm.fulfilmentTimes).toEqual(expectedTimes);
            });
        });

        describe('shouldShowPreOrderWarning', () => {
            let wrapper;

            describe('when asap is not available', () => {
                beforeEach(() => {
                    wrapper = shallowMount(Selector, {
                        store: createStore({
                            ...defaultCheckoutState,
                            availableFulfilment: {
                                times: fulfilmentTimes,
                                isAsapAvailable: false
                            }
                        }),
                        i18n,
                        localVue,
                        propsData
                    });
                });

                it('should return `true`', () => {
                    // Assert
                    expect(wrapper.vm.shouldShowPreOrderWarning).toBeTruthy();
                });

                it('should render warning message', () => {
                    // Arrange
                    const preOrderWarning = wrapper.find('[data-test-id="warning-pre-order"]');

                    // Assert
                    expect(preOrderWarning.exists()).toBeTruthy();
                });
            });

            describe('when asap is available', () => {
                beforeEach(() => {
                    wrapper = shallowMount(Selector, {
                        store: createStore({
                            ...defaultCheckoutState,
                            availableFulfilment: {
                                times: fulfilmentTimes,
                                isAsapAvailable: true
                            }
                        }),
                        i18n,
                        localVue,
                        propsData
                    });
                });

                it('should return `false`', () => {
                    // Assert
                    expect(wrapper.vm.shouldShowPreOrderWarning).toBeFalsy();
                });

                it('should not render warning message', () => {
                    // Arrange
                    const preOrderWarning = wrapper.find('[data-test-id="warning-pre-order"]');

                    // Assert
                    expect(preOrderWarning.exists()).toBeFalsy();
                });
            });
        });
    });

    describe('methods ::', () => {
        describe('selectionChanged', () => {
            const field = 'orderTime';
            const selectedTime = '2020-01-01T01:00:00.000Z';

            let wrapper;
            let updateChangedFieldSpy;
            let setAsapFlagSpy;

            beforeEach(() => {
                updateChangedFieldSpy = jest.spyOn(Selector.methods, 'updateChangedField');
                setAsapFlagSpy = jest.spyOn(Selector.methods, 'setAsapFlag');


                wrapper = shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });
            });

            it('should call `updateChangedField` with `orderTime`', () => {
                // Act
                wrapper.vm.selectionChanged(selectedTime);

                // Assert
                expect(updateChangedFieldSpy).toHaveBeenCalledWith(field);
            });

            it('should call `setAsapFlag` with the `selectedFulfilmentTime`', () => {
                // Act
                wrapper.vm.selectionChanged(selectedTime);

                // Assert
                expect(setAsapFlagSpy).toHaveBeenCalledWith(selectedTime);
            });

            it('should update `selectedAvailableFulfilmentTime` with the value passed', () => {
                // Act
                wrapper.vm.selectionChanged(selectedTime);

                // Assert
                expect(wrapper.vm.selectedAvailableFulfilmentTime).toBe(selectedTime);
            });
        });

        describe('`initFulfilmentTime`', () => {
            let wrapper;
            let updateFulfilmentTimeSpy;

            beforeEach(() => {
                wrapper = shallowMount(Selector, {
                    store: createStore({
                        availableFulfilment: {
                            isAsapAvailable: false,
                            times: []
                        }
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                updateFulfilmentTimeSpy = jest.spyOn(Selector.methods, 'updateFulfilmentTime');
            });

            afterEach(() => {
                jest.clearAllMocks();
            });

            it('should exist', () => {
                expect(wrapper.vm.initFulfilmentTime).toBeDefined();
            });

            describe('when invoked', () => {
                let times;
                beforeEach(() => {
                    // Arrange
                    times = [{
                        text: 'Wednesday 01:00',
                        value: '2020-01-01T01:00:00.000Z'
                    }, {
                        text: 'Wednesday 01:15',
                        value: '2020-01-01T01:15:00.000Z'
                    }, {
                        text: 'Wednesday 01:30',
                        value: '2020-01-01T01:30:00.000Z'
                    }];
                });

                describe('AND fulfilment `times` are supplied, a selected Fulfilment Time is supplied and this time is still available in the fulfilment `times`', () => {
                    // Arrange
                    let selectedFulfilmentTime;

                    beforeEach(() => {
                        // Pre-selected/set fulfilment time
                        selectedFulfilmentTime = {
                            from: '2020-01-01T01:15:00.000Z',
                            to: '2020-01-01T01:15:00.000Z'
                        };

                        // Act
                        wrapper.vm.initFulfilmentTime(times, selectedFulfilmentTime);
                    });

                    it('should update `selectedAvailableFulfilmentTime` with the state Time value', () => {
                        // Assert
                        expect(wrapper.vm.selectedAvailableFulfilmentTime).toEqual(selectedFulfilmentTime.from);
                    });

                    it('should NOT make a call to `updateFulfilmentTime`', () => {
                        // Assert
                        expect(updateFulfilmentTimeSpy).not.toHaveBeenCalled();
                    });
                });

                describe('AND fulfilment `times` are supplied, a selected Fulfilment Time is supplied but this time is now not available in the fulfilment `times`', () => {
                    // Arrange
                    beforeEach(() => {
                        // Arrange
                        const selectedFulfilmentTime = {
                            from: '2020-01-01T00:45:00.000Z',
                            to: '2020-01-01T00:45:00.000Z'
                        };

                        // Act
                        wrapper.vm.initFulfilmentTime(times, selectedFulfilmentTime);
                    });

                    it('should update `selectedAvailableFulfilmentTime` with the first available time value from the fulfilment `times`', () => {
                        // Assert
                        expect(wrapper.vm.selectedAvailableFulfilmentTime).toEqual(times[0].value);
                    });

                    it('should make a call to `updateFulfilmentTime` with the first available `from` & `to` values', () => {
                        // Assert
                        expect(updateFulfilmentTimeSpy).toHaveBeenCalledWith({
                            from: '2020-01-01T01:00:00.000Z',
                            to: '2020-01-01T01:00:00.000Z'
                        });
                    });
                });

                describe('AND no fulfilment `times` are supplied but a selected Fulfilment Time is supplied', () => {
                    beforeEach(() => {
                        const selectedFulfilmentTime = {
                            from: '2020-01-01T01:15:00.000Z',
                            to: '2020-01-01T01:15:00.000Z'
                        };

                        // Act
                        wrapper.vm.initFulfilmentTime([], selectedFulfilmentTime);
                    });

                    it('should NOT make a call to `updateFulfilmentTime`', () => {
                        // Assert
                        expect(updateFulfilmentTimeSpy).not.toHaveBeenCalled();
                    });

                    it('should not update `selectedAvailableFulfilmentTime`', () => {
                        // Assert
                        expect(wrapper.vm.selectedAvailableFulfilmentTime).toEqual('');
                    });
                });

                describe('AND no fulfilment `times` are supplied and no selected Fulfilment Time is supplied', () => {
                    beforeEach(() => {
                        // Act
                        wrapper.vm.initFulfilmentTime([]);
                    });
                    it('should NOT make a call to `updateFulfilmentTime`', () => {
                        // Assert
                        expect(updateFulfilmentTimeSpy).not.toHaveBeenCalled();
                    });

                    it('should not update `selectedAvailableFulfilmentTime`', () => {
                        // Assert
                        expect(wrapper.vm.selectedAvailableFulfilmentTime).toEqual('');
                    });
                });

                describe('AND fulfilment `times` are supplied but no selected Fulfilment Time is supplied', () => {
                    beforeEach(() => {
                        // Act
                        wrapper.vm.initFulfilmentTime(times);
                    });

                    it('should make a call to `updateFulfilmentTime` with the `from` & `to` values', () => {
                        // Assert
                        expect(updateFulfilmentTimeSpy).toHaveBeenCalledWith({
                            from: '2020-01-01T01:00:00.000Z',
                            to: '2020-01-01T01:00:00.000Z'
                        });
                    });

                    it('should update `selectedAvailableFulfilmentTime` with the first available fulfilment Time value', () => {
                        // Assert
                        expect(wrapper.vm.selectedAvailableFulfilmentTime).toEqual(times[0].value);
                    });
                });
            });
        });

        describe('`setAsapFlag`', () => {
            let wrapper;
            let updateHasAsapSelectedSpy;

            beforeEach(() => {
                wrapper = shallowMount(Selector, {
                    store: createStore({
                        availableFulfilment: {
                            isAsapAvailable: true,
                            times: fulfilmentTimes
                        }
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                updateHasAsapSelectedSpy = jest.spyOn(Selector.methods, 'updateHasAsapSelected');
            });

            it('should exist', () => {
                expect(wrapper.vm.setAsapFlag).toBeDefined();
            });

            describe('when invoked', () => {
                describe('AND `isAsapAvailable` is available & the `availableFulfilment` from matches the `selected` fulfilment time', () => {
                    it('should make a call to `updateHasAsapSelected` with the param `true` passed', () => {
                        // Act
                        wrapper.vm.setAsapFlag();

                        // Assert
                        expect(updateHasAsapSelectedSpy).toHaveBeenCalledWith(true);
                    });
                });

                describe('AND `isAsapAvailable` is not available', () => {
                    it('should make a call to `updateHasAsapSelected` with the param `false` passed', () => {
                        // Act
                        wrapper.vm.setAsapFlag();

                        // Assert
                        expect(updateHasAsapSelectedSpy).toHaveBeenCalledWith(false);
                    });
                });
            });
        });
    });

    describe('watch ::', () => {
        describe('fulfilmentTimes ::', () => {
            afterEach(() => {
                jest.clearAllMocks();
            });
            it('should call `selectionChanged` with the first fulfilment time when there are fulfilment times', () => {
                // Arrange
                const expected = '2020-01-01T01:00:00.000Z';
                const selectionChangedSpy = jest.spyOn(Selector.methods, 'selectionChanged');

                const wrapper = shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });

                // Act
                wrapper.vm.$options.watch.fulfilmentTimes.call(wrapper.vm, [
                    {
                        text: 'Wednesday 01:00',
                        value: '2020-01-01T01:00:00.000Z'
                    }
                ]);

                // Assert
                expect(selectionChangedSpy).toHaveBeenCalledWith(expected);
            });

            it('should call `selectionChanged` with `undefined` when there are no fulfilment times', () => {
                // Arrange
                const selectionChangedSpy = jest.spyOn(Selector.methods, 'selectionChanged');

                const wrapper = shallowMount(Selector, {
                    store: createStore({
                        ...defaultCheckoutState,
                        availableFulfilment: {
                            isAsapAvailable: false,
                            times: []
                        },
                        serviceType: CHECKOUT_METHOD_DELIVERY
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                // Act
                wrapper.vm.$options.watch.fulfilmentTimes.call(wrapper.vm, []);

                // Assert
                expect(selectionChangedSpy).toHaveBeenCalledWith(undefined);
            });
        });
    });

    describe('mounted ::', () => {
        describe('Initial Page Visit ::', () => {
            it('should make a call to `initFulfilmentTime` with the available `fulfilmentTimes` and no preset `fulfilmentTime`', () => {
                // Arrange
                const initFulfilmentTimeSpy = jest.spyOn(Selector.methods, 'initFulfilmentTime');

                // Act
                shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });

                // Assert
                expect(initFulfilmentTimeSpy).toHaveBeenCalledWith(
                    [
                        {
                            text: 'As soon as possible',
                            value: '2020-01-01T01:00:00.000Z'
                        },
                        {
                            text: 'Wednesday 01:15',
                            value: '2020-01-01T01:15:00.000Z'
                        }
                    ],
                    {
                        from: '',
                        to: ''
                    }
                );
            });
        });

        describe('Page Re-visit ::', () => {
            it('should make a call to `initFulfilmentTime` with the available `fulfilmentTimes` and any preset `fulfilmentTime`', () => {
                const preselectedTime = {
                    from: '2020-01-01T01:15:00.000Z',
                    to: '2020-01-01T01:15:00.000Z'
                };
                defaultCheckoutState.time = preselectedTime;

                // Arrange
                const initFulfilmentTimeSpy = jest.spyOn(Selector.methods, 'initFulfilmentTime');

                // Act
                shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });

                // Assert
                expect(initFulfilmentTimeSpy).toHaveBeenCalledWith(
                    [
                        {
                            text: 'As soon as possible',
                            value: '2020-01-01T01:00:00.000Z'
                        },
                        {
                            text: 'Wednesday 01:15',
                            value: '2020-01-01T01:15:00.000Z'
                        }
                    ],
                    preselectedTime
                );
            });
        });
    });
});
