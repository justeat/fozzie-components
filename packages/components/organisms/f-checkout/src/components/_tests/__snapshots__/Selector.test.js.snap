// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector computed :: orderMethod should show the collection label when the \`serviceType\` is \`collection\` 1`] = `
<select id="formField-order-time" data-test-id="formfield-order-time-dropdown-select" name="order-time" class="FormDropdown_c-formDropdown-select_ROvjN">
  <option data-test-id="formfield-order-time-dropdown-option-0" value="[object Object]">{
    "text": "As soon as possible",
    "value": "2020-01-01T01:00:00.000Z"
    }</option>
  <option data-test-id="formfield-order-time-dropdown-option-1" value="[object Object]">{
    "text": "Wednesday 01:15",
    "value": "2020-01-01T01:15:00.000Z"
    }</option>
</select>
`;

exports[`Selector computed :: orderMethod should show the delivery label when the \`serviceType\` is \`delivery\` 1`] = `
<select id="formField-order-time" data-test-id="formfield-order-time-dropdown-select" name="order-time" class="FormDropdown_c-formDropdown-select_ROvjN">
  <option data-test-id="formfield-order-time-dropdown-option-0" value="[object Object]">{
    "text": "As soon as possible",
    "value": "2020-01-01T01:00:00.000Z"
    }</option>
  <option data-test-id="formfield-order-time-dropdown-option-1" value="[object Object]">{
    "text": "Wednesday 01:15",
    "value": "2020-01-01T01:15:00.000Z"
    }</option>
</select>
`;
